### PR TITLE
Add docs for diffusers autologger

### PR DESCRIPTION
**Pull request by soumik12345**
## Description

The PR adds documentation for the HuggingFace Diffusers autologger shipped as part of https://github.com/wandb/wandb/pull/6561.

https://github.com/wandb/docodile/assets/19887676/6977c8b8-4eaf-46d6-b400-8b1e2fbb7d99

## Checklist

Check if your PR fulfills the following requirements. Put an `X` in the boxes that apply.

- [x] Files I edited were previewed on my local development server with `yarn start`. My changes did not break the local preview.
- [ ] Build (`yarn docusaurus build`) was run locally and successfully without errors or warnings.
- [ ] I merged the latest changes from `main` into my feature branch before submitting this PR.

### CODE REVIEW

File: docs/guides/integrations/diffusers.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Hugging Face Diffusers
+
+🤗 Diffusers is the go-to library for state-of-the-art pretrained diffusion models for generating images, audio, and even 3D structures of molecules. Whether you’re looking for a simple inference solution or want to train your own diffusion model, 🤗 Diffusers is a modular toolbox that supports both. 🤗 Diffusers is designed with a focus on [usability over performance](https://huggingface.co/docs/diffusers/v0.23.1/en/conceptual/philosophy#usability-over-performance), [simple over easy](https://huggingface.co/docs/diffusers/v0.23.1/en/conceptual/philosophy#simple-over-easy), and [customizability over abstractions](https://huggingface.co/docs/diffusers/v0.23.1/en/conceptual/philosophy#tweakable-contributorfriendly-over-abstraction).
+
+## Getting Started
```
Comment (morganmcg1) [inline docs/guides/integrations/diffusers.md:55]: to start with,  just show the autolog and showing calling a pipeline class, similar to the transformers example:

https://docs.wandb.ai/guides/integrations/huggingface

File: docs/guides/integrations/diffusers.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+# Define the prompts, negative prompts, and seed.
+prompt = [
+    "a photograph of an astronaut riding a horse",
+    "a photograph of a dragon"
+]
+negative_prompt = ["ugly, deformed", "ugly, deformed"]
+generator = torch.Generator(device="cpu").manual_seed(10)
+
+# call the autologger before calling the pipeline
+autolog(init=dict(project="diffusers_logging"))
```
Comment (morganmcg1) [inline docs/guides/integrations/diffusers.md]: lets encourage users to call this at the very start of their script, under the imports

File: docs/guides/integrations/diffusers.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
+---
+displayed_sidebar: default
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Hugging Face Diffusers
+
+🤗 Diffusers is the go-to library for state-of-the-art pretrained diffusion models for generating images, audio, and even 3D structures of molecules. Whether you’re looking for a simple inference solution or want to train your own diffusion model, 🤗 Diffusers is a modular toolbox that supports both. 🤗 Diffusers is designed with a focus on [usability over performance](https://huggingface.co/docs/diffusers/v0.23.1/en/conceptual/philosophy#usability-over-performance), [simple over easy](https://huggingface.co/docs/diffusers/v0.23.1/en/conceptual/philosophy#simple-over-easy), and [customizability over abstractions](https://huggingface.co/docs/diffusers/v0.23.1/en/conceptual/philosophy#tweakable-contributorfriendly-over-abstraction).
```
Comment (morganmcg1) [inline docs/guides/integrations/diffusers.md]: "Whether you’re looking for a simple inference solution or want to train your own diffusion model, 🤗 Diffusers is a modular toolbox that supports both. 🤗 Diffusers is designed with a focus on [usability over performance](https://huggingface.co/docs/diffusers/v0.23.1/en/conceptual/philosophy#usability-over-performance), [simple over easy](https://huggingface.co/docs/diffusers/v0.23.1/en/conceptual/philosophy#simple-over-easy), and [customizability over abstractions](https://huggingface.co/docs/diffusers/v0.23.1/en/conceptual/philosophy#tweakable-contributorfriendly-over-abstraction)"

 - delete, not critical to understanding diffusers or wandb. Instead describe why a user should use the integration and what they can get out of it.

Comment (morganmcg1) [reply docs/guides/integrations/diffusers.md]: Maybe also put 1 of the nice gifs up here towards the top

File: docs/guides/integrations/diffusers.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+```python
+import torch
+from diffusers import DiffusionPipeline
+from wandb.integration.diffusers import autolog
+
+
+# Initialize the diffusion pipeline
+pipeline = DiffusionPipeline.from_pretrained(
+    "runwayml/stable-diffusion-v1-5", torch_dtype=torch.float16
```
Comment (morganmcg1) [inline docs/guides/integrations/diffusers.md]: maybe highlight a smaller/faster/more modern model? does it work with LCM?

File: docs/guides/integrations/diffusers.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+# call the pipeline to generate the images
+images = pipeline(
+    prompt,
+    negative_prompt=negative_prompt,
+    num_images_per_prompt=2,
+    generator=generator,
+    guidance_rescale=0.0,
+)
+```
```
Comment (morganmcg1) [inline docs/guides/integrations/diffusers.md:112]: Add a line or 2 summarising what has just happened and where the users can find their logs.

File: docs/guides/integrations/diffusers.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+| ![This gif shows how the results of your experiment are logged](@site/static/images/integrations/diffusers-autolog-2.gif) |
```
Comment (morganmcg1) [inline docs/guides/integrations/diffusers.md]: Add a section, maybe here, with a table with all arguments to the autolog, similar to the HF transformers section here: https://docs.wandb.ai/guides/integrations/huggingface#additional-wb-settings

File: docs/guides/integrations/diffusers.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+|:--:| 
+| **This gif shows how your workspace would look with multiple experiments.** |
+
+| ![This gif shows how the autologger logs the configs of your experiment](@site/static/images/integrations/diffusers-autolog-3.gif) | 
+|:--:| 
+| **This gif shows how the autologger logs the configs of your experiment.** |
+
+You can find a list of supported pipeline calls [here](https://github.com/wandb/wandb/blob/main/wandb/integration/diffusers/autologger.py#L12-L67). In case, you want to request a new feature of this integration or report a bug associated with it, please open an issue on [https://github.com/wandb/wandb/issues](https://github.com/wandb/wandb/issues).
+
+## More Resources
```
Comment (morganmcg1) [inline docs/guides/integrations/diffusers.md:136]: nice!

File: docs/guides/integrations/diffusers.md (hunk 2 above)
Comment (soumik12345) [reply docs/guides/integrations/diffusers.md]: Although I updated the docs to call the `autolog` at the start of the program, in practice, I would discourage this. Ideally, you would want to call `autolog` after you have instantiated your pipeline, because in case you want to do back-to-back experiments with a pipeline, you wouldn't need to reload the pipeline or at the worst case restart the pipeline. Check the code for [this HF Space](https://huggingface.co/spaces/wandb/reproducible-pixart-alpha/blob/main/app.py).

File: docs/guides/integrations/diffusers.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+```
+
+Calling the `autolog` creates a [W&B run](https://docs.wandb.ai/guides/runs). All subsequent pipeline calls are therefore registered and their inputs and outputs are logged to the run. The prompts, negative prompts, and the generated media are logged in a [`wandb.Table`](https://docs.wandb.ai/guides/tables) while all other configs associated with the experiment including seed and the pipeline architecture are stored in the config section for the run. The generated media are also logged in [media panels](https://docs.wandb.ai/guides/track/log/media) in the run.
+
+| ![An example of how the results of your experiment are logged](@site/static/images/integrations/diffusers-autolog-4.gif) | 
+|:--:| 
+| **An example of how the results of your experiment are logged.** |
+
+:::info
+The arguments passed to the `autolog` function is just `init` which accepts a dictionary of keywork arguments that would be passed to [`wandb.init()`](https://docs.wandb.ai/ref/python/init).
```
Comment (morganmcg1) [inline docs/guides/integrations/diffusers.md]: typo, "keywork"

File: docs/guides/integrations/diffusers.md [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+
+| ![An example of how the results of your experiment are logged](@site/static/images/integrations/diffusers-autolog-1.gif) | 
+|:--:| 
+| **An example of how the results of multiple experiments are logged in your workspace.** |
+
+| ![An example of how the autologger logs the configs of your experiment](@site/static/images/integrations/diffusers-autolog-3.gif) | 
+|:--:| 
+| **An example of how the autologger logs the configs of your experiment.** |
+
+:::info
```
Comment (morganmcg1) [inline docs/guides/integrations/diffusers.md:132]: nice

**Review by morganmcg1 — APPROVED** [review]

**Review by ngrayluna — APPROVED** [review]
